controller/src/main/java/org/jboss/as/controller/access/JmxTarget.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2015 Red Hat, inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301  USA
+ */
+package org.jboss.as.controller.access;
+
+import javax.management.ObjectName;
+
+/**
+ *
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a> (c) 2015 Red Hat, inc.
+ */
+public class JmxTarget {
+
+    private final String method;
+    private final ObjectName objectName;
+    private boolean isNonFacadeMBeansSensitive;
+
+    public JmxTarget(String method, ObjectName objectName, boolean isNonFacadeMBeansSensitive) {
+        this.method = method;
+        this.objectName = objectName;
+        this.isNonFacadeMBeansSensitive = isNonFacadeMBeansSensitive;
+    }
+
+    public String getMethod() {
+        return method;
+    }
+
+    public ObjectName getObjectName() {
+        return objectName;
+    }
+
+    public boolean isNonFacadeMBeansSensitive() {
+        return isNonFacadeMBeansSensitive;
+    }
+}
```

Comment:
This should provide these:

HostEffect getHostEffect()

ServerGroupEffect getServerGroupEffect()

Those can then drive HostEffectConstraint and ServerGroupEffectConstraint.

The trick then is to get the name of the host and the name of the server group in to PluggableMBeanServerImpl.

Possibly the HostEffect and ServerGroupEffect themselves can be provided instead of the details used to create them. Passing them via JmxAuthorizer is a simple solution.

On a domain server, the host name and server-group name are recorded in the root management resource. They are undefined of course in a standalone server. We should also check with Kabir Khan as to whether this code should function in an HC process as well (IIRC, his plan is to have the JMX subsystem function as a HC subsystem too), in which case, the host name is known in the model too, while the server group name is undefined.